node/flatpak-node-generator.py

```diff
@@ -1405,14 +1385,12 @@ async def generate_package(self, package: Package) -> None:
 
         elif isinstance(source, GitSource):
             repo_name = urllib.parse.urlparse(source.url).path.split('/')[-1]
-            tarball_url = source.tarball_url
-            assert tarball_url is not None
-            metadata = await RemoteUrlMetadata.get(tarball_url, cachable=True)
-            filename = f'{repo_name}-{source.commit}'
-
-            #XXX Yarn wants uncompressed tar, why?
-            self.gen.add_url_source(tarball_url, metadata.integrity, self.mirror_dir / f'{filename}.gz')
-            self.gen.add_command(f'gunzip "{self.mirror_dir}/{filename}.gz"')
+            name = f'{repo_name}-{source.commit}'
+            repo_dir = self.gen.tmp_root / name
+            target_tar = self.mirror_dir / name
+
+            self.gen.add_git_source(source.url, source.commit, repo_dir)
+            self.gen.add_command(f'cd {repo_dir}; git archive --format tar -o ../../../{target_tar} HEAD')
```

Comment:
It's more of a personal preference, but I'd rather calculate relative path at python level, e.g.
```suggestion
            repo_dir = self.gen.tmp_root / name
            target_tar = os.path.relpath(self.mirror_dir / name, repo_dir)

            self.gen.add_git_source(source.url, source.commit, repo_dir)
            self.gen.add_command(f'cd {repo_dir}; git archive --format tar -o {target_tar} HEAD')
```

Reply:
Adressed.